graylog2-web-interface/test/helpers/suppressConsole.ts

```diff
@@ -23,14 +23,17 @@
  *
  * @param {function} fn - the function which should be called after disabling `console.error` and before restoring it.
  */
-const suppressConsole = (fn: () => void) => {
+const suppressConsole = <T extends () => any>(fn: T): Promise<ReturnType<T>> => {
   /* eslint-disable no-console */
   const originalConsoleError = console.error;
   console.error = () => {};
 
-  fn();
+  return Promise.resolve(fn()).then((result) => {
```

Comment:
What happens when the promise is rejected? From what I can see, `console.error` will not be reset in that case. Maybe use `finally`?

Reply:
Good point.